pkg/daemon/daemon.go

```diff
@@ -131,57 +119,107 @@ func applyNodePTPProfile(pm *ProcessManager, nodeProfile *ptpv1.PtpProfile) erro
+
+	glog.Infof("------------------------------------")
+	printWhenNotNil(nodeProfile.Name, "Profile Name")
+	printWhenNotNil(nodeProfile.Interface, "Interface")
+	printWhenNotNil(nodeProfile.Ptp4lOpts, "Ptp4lOpts")
+	printWhenNotNil(nodeProfile.Ptp4lConf, "Ptp4lConf")
+	printWhenNotNil(nodeProfile.Phc2sysOpts, "Phc2sysOpts")
+	glog.Infof("------------------------------------")
+
 	if nodeProfile.Phc2sysOpts != nil {
-		pm.process = append(pm.process, &ptpProcess{
+		dn.processManager.process = append(dn.processManager.process, &ptpProcess{
 			name:   "phc2sys",
+			iface:  *nodeProfile.Interface,
 			exitCh: make(chan bool),
 			cmd:    phc2sysCreateCmd(nodeProfile)})
 	} else {
-		glog.Infof("applyNodePTPProfile: not starting phc2sys, phc2sysOpts is empty")
+		glog.Infof("applyNodePtpProfile: not starting phc2sys, phc2sysOpts is empty")
 	}
 
 	if nodeProfile.Ptp4lOpts != nil && nodeProfile.Interface != nil {
-		if nodeProfile.Ptp4lConf != nil && *nodeProfile.Ptp4lConf != "" {
-			if _, err := os.Stat(PTP4L_CONF_DIR); os.IsNotExist(err) {
-				err = os.Mkdir(PTP4L_CONF_DIR, 0644)
-				if err != nil {
-					return fmt.Errorf("failed to create the ptp4l custom configuration directory: %v", err)
-				}
-			}
-
-			err := ioutil.WriteFile(fmt.Sprintf("%s/%s.conf", PTP4L_CONF_DIR, *nodeProfile.Name), []byte(*nodeProfile.Ptp4lConf), 0644)
-			if err != nil {
-				return fmt.Errorf("failed to write the configuration file named %s: %v", fmt.Sprintf("%s/%s.conf", PTP4L_CONF_DIR, *nodeProfile.Name), err)
-			}
+		configPath := fmt.Sprintf("/var/run/ptp4l.%d.config", runID)
+		err := ioutil.WriteFile(configPath, []byte(*nodeProfile.Ptp4lConf), 0644)
+		if err != nil {
+			return fmt.Errorf("failed to write the configuration file named %s: %v", configPath, err)
 		}
 
-		pm.process = append(pm.process, &ptpProcess{
-			name:   "ptp4l",
-			exitCh: make(chan bool),
-			cmd:    ptp4lCreateCmd(nodeProfile)})
+		dn.processManager.process = append(dn.processManager.process, &ptpProcess{
+			name:            "ptp4l",
+			iface:           *nodeProfile.Interface,
+			ptp4lConfigPath: configPath,
+			ptp4lSocketPath: socketPath,
+			exitCh:          make(chan bool),
+			cmd:             ptp4lCreateCmd(nodeProfile, configPath)})
 	} else {
-		glog.Infof("applyNodePTPProfile: not starting ptp4l, ptp4lOpts or interface is empty")
+		glog.Infof("applyNodePtpProfile: not starting ptp4l, ptp4lOpts or interface is empty")
 	}
 
-	for _, p := range pm.process {
-		if p != nil {
-			time.Sleep(1 * time.Second)
-			go cmdRun(p)
-		}
-	}
 	return nil
 }
 
+func (dn *Daemon) addProfileConfig(runID int, nodeProfile *ptpv1.PtpProfile) string {
+	// TODO: later implement a merge capability
+	if nodeProfile.Ptp4lConf == nil || *nodeProfile.Ptp4lConf == "" {
+		// We need to copy this to another variable because is a pointer
+		config := string(dn.ptpUpdate.defaultPTP4lConfig)
+		nodeProfile.Ptp4lConf = &config
+	}
+
+	socketPath := fmt.Sprintf("/var/run/ptp4l.%d.socket", runID)
```

Comment:
Maybe define both socketPath outside of addProfileConfig and pass it in as a param?
It doesn't sound right that addProfileConfig returns a socket path which is only used by phc2sys instance.